File: csharp/MetaField.cs
namespace MetaObjects;

/// <summary>Concrete node — a field.</summary>
public class MetaField : MetaData
{
    public MetaField(string subType, string name) : base("field", subType, name) { }

    /// <summary>Effective validators — own + super-chain-inherited.</summary>
    public IReadOnlyList<MetaData> Validators() =>
        Cached("validators", () =>
            (IReadOnlyList<MetaData>)EffectiveChildren().Where(c => c.Type == "validator").ToList());
}

File: server/python/src/metaobjects/meta_field.py
"""Concrete node — a field."""
from __future__ import annotations

from .meta_data import MetaData


class MetaField(MetaData):
    def __init__(self, sub_type: str, name: str) -> None:
        super().__init__("field", sub_type, name)

    def validators(self) -> list[MetaData]:
        """Effective validators — own + super-chain-inherited."""
        return self._cached(
            "validators",
            lambda: [c for c in self.effective_children() if c.type == "validator"],
        )
